_rules/SC4-1-2-iframe-has-name.md

```diff
@@ -0,0 +1,149 @@
+---
+name: iframe has an accessible name
+
+description: |
+ Each iframe element has an accessible name
+
+success_criterion:
+- 4.1.2 # Name, Role, Value (A)
+
+test_aspects:
+- DOM Tree
+- CSS Styling
+
+authors:
+- Jey Nandakumar
+---
+
+## Test procedure
+
+### Applicability
+
+The rule applies to `iframe` elements that are [exposed to assistive technologies](#exposed-to-assistive-technologies).
+
+**Note:** `frame` element is deprecated, this rule does not consider `frame` or `frameset` elements.
+
+### Expectation
+
+Each target element has an [accessible name](#accessible-name) that is [non-empty](#non-empty).
+
+## Assumptions
+
+- The rule assumes that the target `iframe` is used as an user interface component.
```

Comment:
Do we need this assumption given that is the applicability? Not sure.

Reply:
I think we do, but I suggest we link to the definition of "user interface component" in WCAG 2.1. @nitedog the reason for this is because 4.1.2 is only applicable to user interface components. This is a highly subjective term, and there could be scenarios where iframes are not distinguishable as its own component. If that's the case, 4.1.2 doesn't apply.